client/src/modules/stock/lots-duplicates/duplicate-lots.js

```diff
@@ -0,0 +1,165 @@
+      headerTooltip : 'STOCK.INITIAL_QUANTITY',
+      headerCellFilter : 'translate',
+      cellClass : 'text-right',
+      type : 'number',
+    }, {
+      field : 'unit_cost',
+      displayName : 'TABLE.COLUMNS.UNIT_PRICE',
+      headerTooltip : 'TABLE.COLUMNS.UNIT_PRICE',
+      headerCellFilter : 'translate',
+      cellFilter : `currency:${vm.enterprise.currency_id}`,
+      cellClass : 'text-right',
+      type : 'number',
+    }, {
+      field : 'entry_date',
+      displayName : 'STOCK.ENTRY_DATE',
+      headerTooltip : 'STOCK.ENTRY_DATE',
+      headerCellFilter : 'translate',
+      cellFilter : 'date',
+      cellClass : 'text-right',
+    }, {
+      field : 'expiration_date',
+      displayName : 'STOCK.EXPIRATION_DATE',
+      headerTooltip : 'STOCK.EXPIRATION_DATE',
+      headerCellFilter : 'translate',
+      cellFilter : 'date:"mediumDate"',
+      cellClass : 'text-right',
+    }, {
+      field : 'num_duplicates',
+      displayName : 'TABLE.COLUMNS.NUM_DUPLICATE_LOTS',
+      headerTooltip : 'TABLE.COLUMNS.NUM_DUPLICATE_LOTS',
+      headerCellFilter : 'translate',
+      cellClass : 'text-right',
+      type : 'number',
+    }, {
+      field : 'action',
+      displayName : '',
+      enableFiltering : false,
+      enableSorting : false,
+      cellTemplate : 'modules/stock/lots-duplicates/templates/action.cell.html',
+    },
+  ];
+
+  const footerTemplate = `
+    <div class="ui-grid-cell-contents">
+      <b>{{ grid.appScope.gridOptions.data.length }}</b>
+      <span translate>FORM.INFO.FOUND</span>
+    </div>
+  `;
+
+  vm.gridOptions = {
+    appScopeProvider   : vm,
+    enableColumnMenus  : false,
+    columnDefs         : columns,
+    showGridFooter     : true,
+    gridFooterTemplate : footerTemplate,
+    fastWatch          : true,
+    flatEntityAccess   : true,
+    onRegisterApi,
+  };
+
+  const gridColumns = new Columns(vm.gridOptions, cacheKey);
+  const state = new GridState(vm.gridOptions, cacheKey);
+
+  vm.gridApi = {};
+  vm.saveGridState = state.saveGridState;
+
+  function clearGridState() {
+    state.clearGridState();
+    $state.reload();
+  }
+
+  function onRegisterApi(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  // expose view logic
+  vm.openColumnConfigModal = openColumnConfigModal;
+  vm.clearGridState = clearGridState;
+
+  // This function opens a modal through column service to let the user toggle
+  // the visibility of the inventories registry's columns.
+  function openColumnConfigModal() {
+    // column configuration has direct access to the grid API to alter the current
+    // state of the columns - this will be saved if the user saves the grid configuration
+    gridColumns.openConfigurationModal();
+  }
+
+  // load stock lots in the grid
+  function load() {
+    vm.hasError = false;
+    vm.loading = true;
+
+    Lots.dupes({ find_dupes : true })
+      .then((rows) => {
+        vm.gridOptions.data = rows;
+      })
+      .catch(Notify.handleError)
```

Comment:
You'll want to set "hasError" to true here.
```js
.catch((err) => {
  hasError = true;
  Notify.handleError(err);
})
```

This tells the grid to show an error message.  `Notify.handleError` doesn't do that part automatically - it just makes the notification bar come down.

Reply:
Updated